### PR TITLE
fix (TDP-4027) fix integer overflow in NumberHistogram stream

### PR DESCRIPTION
**Pull request by Vince-J**
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-4027
**Please check if the PR fulfills these requirements**
- [X] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [X] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [X] The new code does not introduce new technical issues (sonar / eslint)
- [X] Functional tests have been performed
- [ ] Docker configuration files for config-std or config-cloud profiles are impacted

**Please check the browsers you've tested on**
- [ ] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [X] No, and no need to (backend changes only)

### CODE REVIEW

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/578/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/578/)

File: dataprep-backend-common/src/main/java/org/talend/dataprep/api/dataset/statistics/number/StreamNumberHistogramStatistics.java [hunk 1]
```diff
@@ -236,21 +240,33 @@ private void extendToRight(double d) {
...
                 count += regulars[j];
                 regulars[j] = 0;
             }
+
             regulars[k++] = count;
+
+            // because i is an integer and can be overflowed if i+factor > Integer.MAX_VALUE, let's use a long temp variable
+            long temp = i + factor;
+            if (temp < numberOfBins) {
+                i += factor;
```
Comment (bdiouf) [inline dataprep-backend-common/src/main/java/org/talend/dataprep/api/dataset/statistics/number/StreamNumberHistogramStatistics.java]: IMHO, even if in this case, we, for sure, know ( factor < numberOfBins) that the assignment will work it is more understandable to write it like this:  **i += (int) factor;**

Comment (vjaouen) [inline dataprep-backend-common/src/main/java/org/talend/dataprep/api/dataset/statistics/number/StreamNumberHistogramStatistics.java]: I agree with you. Change done

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/584/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/584/)

**Pull request merged**